SpringKonstant/src/org/usfirst/frc6357/robotcode/subsystems/TFMini.java

```diff
@@ -111,7 +111,11 @@ public void stop()
      */
```

Comment:
Here and in a few more places you'll see a synchronized(this) block. Essentially, this is a mutex (mutual exclusion semaphore) on this (the self-reference). When execution reaches a synchronized block, it checks to see if any other thread is in a synchronized block on the same reference object. If there is another thread in a synchronized block, then the new execution yields to that block. Once complete, execution returns to the blocked synchronize.

This mechanism prevents multiple read and write operations from being interrupted. It also prevents a variable from being read if it is being written to at the same time.